js/jquery.flot.tooltip.source.js

```diff
@@ -154,22 +154,27 @@
                     if (self.tooltips[STICKY_TOOLTIP]) {
                         hideTooltip(STICKY_TOOLTIP);
                     }
-                    // there is a tooltip node already attached to the plot since the user has
-                    // to hover over the point before clicking, so we reassign it here instead of
-                    // making a new one
-                    // then we remove the hover tooltip entry to clean up the internal state
-                    self.tooltips[STICKY_TOOLTIP] = self.tooltips[HOVER_TOOLTIP];
-                    self.tooltips[STICKY_TOOLTIP].item = item;
-                    delete self.tooltips[HOVER_TOOLTIP];
-
-                    self.plot.highlight(item.seriesIndex, item.dataIndex);
+                    // In most cases, there is a tooltip node already attached to the plot since
+                    // the user has to hover over the point before clicking, so we reassign it
+                    // here instead of making a new one
+                    // Then we remove the hover tooltip entry to clean up the internal state
+                    // However, there will be no hover tooltip if the user clicks the same point
+                    // to unsticky a tooltip since we do not duplicate hovered/sticky tooltips
+                    if (self.tooltips[HOVER_TOOLTIP]) {
+                        self.tooltips[STICKY_TOOLTIP] = self.tooltips[HOVER_TOOLTIP];
+                        self.tooltips[STICKY_TOOLTIP].item = item;
+                        delete self.tooltips[HOVER_TOOLTIP];
+
+                        self.plot.highlight(item.seriesIndex, item.dataIndex);
+
+                        const $tip = self.tooltips[STICKY_TOOLTIP].node;
+                        $tip.addClass(self.tooltipOptions.stickyClass);
+                        if (typeof self.tooltipOptions.onClick === 'function') {
+                            self.tooltipOptions.onClick.call(self.plot, item, $tip, true);
+                        }
+                    }
```

Comment:
I don't think this issue is completely fixed. As I repeatedly click on the same point with my mouse I don't see the tooltip toggle between stick and unstick state, rather the tooltip disappears after first click and the dot comes up in the second click and all subsequent clicks have no effect.

It looks like the stick state is not cleared out completely because I see the vertical line still fixed to the same point even after I unstick the tooltip with a second click.

Reply:
I updated the docker with a fix for this